File: frontend/src/Service/CategoryService.js
import axios from "axios";

const API_URL = "http://localhost:5234/api/Admin/Category"; 

const CategoryService = {
  getAllCategories: () => axios.get(API_URL),
  getCategoryById: (id) => axios.get(`${API_URL}/${id}`),
  
  createCategory: async (data) => {
    try {
      const response = await axios.post(API_URL, data);
      return response.data;
    } catch (error) {
      if (error.response && error.response.data && error.response.data.message) {
        throw new Error(error.response.data.message);
      } else {
        throw new Error('Có lỗi xảy ra khi thêm danh mục');
      }
    }
  },
  
  updateCategory: async (id, data) => {
    try {
      const response = await axios.put(`${API_URL}/${id}`, data);
      return response.data;
    } catch (error) {
      if (error.response && error.response.data && error.response.data.message) {
        throw new Error(error.response.data.message);
      } else {
        throw new Error('Có lỗi xảy ra khi cập nhật danh mục');
      }
    }
  },
  
  deleteCategory: (id) => axios.delete(`${API_URL}/${id}`)
};

export default CategoryService;

File: doancnpmnv_flutter/lib/services/category_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class CategoryService {
  static const String baseUrl = "http://10.0.2.2:5234/api/Admin/Category";

  // Lấy tất cả danh mục
  static Future<List<dynamic>> getAllCategories() async {
    final response = await http.get(Uri.parse(baseUrl));

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception("Không thể tải danh mục");
    }
  }

  // Thêm danh mục
  static Future<String> addCategory(String name) async {
    final response = await http.post(
      Uri.parse(baseUrl),
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({"name": name}),
    );

    if (response.statusCode == 201) {
      return "Thêm danh mục thành công!";
    } else {
      final data = jsonDecode(response.body);
      return data['message'] ?? "Lỗi khi thêm danh mục";
    }
  }

  // Cập nhật danh mục
  static Future<String> updateCategory(int id, String name) async {
    final response = await http.put(
      Uri.parse("$baseUrl/$id"),
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({"categoryID": id, "name": name}),
    );

    if (response.statusCode == 204) {
      return "Cập nhật thành công!";
    } else {
      final data = jsonDecode(response.body);
      return data['message'] ?? "Lỗi khi cập nhật danh mục";
    }
  }

  // Xóa danh mục
  static Future<String> deleteCategory(int id) async {
    final response = await http.delete(Uri.parse("$baseUrl/$id"));

    if (response.statusCode == 204) {
      return "Xóa danh mục thành công!";
    } else {
      final data = jsonDecode(response.body);
      return data['message'] ?? "Không thể xóa danh mục.";
    }
  }
}
